Hmm, seems the the label is still mandatory:

```log
homeassistant/components/shelly/config_flow.py:51: error: Missing key "label" for TypedDict "SelectOptionDict"  [typeddict-item]
homeassistant/components/shelly/config_flow.py:52: error: Missing key "label" for TypedDict "SelectOptionDict"  [typeddict-item]
homeassistant/components/shelly/config_flow.py:53: error: Missing key "label" for TypedDict "SelectOptionDict"  [typeddict-item]
```

--- a/homeassistant/components/shelly/config_flow.py
+++ b/homeassistant/components/shelly/config_flow.py
@@ -22,7 +22,6 @@ from homeassistant.core import HomeAssistant, callback
 from homeassistant.data_entry_flow import FlowResult
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
 from homeassistant.helpers.selector import (
-    SelectOptionDict,
     SelectSelector,
     SelectSelectorConfig,
 )
@@ -53,9 +52,9 @@ HOST_SCHEMA: Final = vol.Schema({vol.Required(CONF_HOST): str})
 
 
 BLE_SCANNER_OPTIONS = [
-    SelectOptionDict(value=BLEScannerMode.DISABLED, label="Disabled"),
-    SelectOptionDict(value=BLEScannerMode.ACTIVE, label="Active"),
-    SelectOptionDict(value=BLEScannerMode.PASSIVE, label="Passive"),
+    BLEScannerMode.DISABLED,
+    BLEScannerMode.ACTIVE,
+    BLEScannerMode.PASSIVE,
 ]
 
 INTERNAL_WIFI_AP_IP = "192.168.33.1"

--- a/homeassistant/components/shelly/config_flow.py
+++ b/homeassistant/components/shelly/config_flow.py
@@ -21,10 +21,7 @@ from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.data_entry_flow import FlowResult
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
-from homeassistant.helpers.selector import (
-    SelectSelector,
-    SelectSelectorConfig,
-)
+from homeassistant.helpers.selector import SelectSelector, SelectSelectorConfig
 
 from .const import (
     BLE_MIN_VERSION,

--- a/homeassistant/components/shelly/config_flow.py
+++ b/homeassistant/components/shelly/config_flow.py
@@ -403,7 +403,10 @@ class OptionsFlowHandler(OptionsFlow):
                             CONF_BLE_SCANNER_MODE, BLEScannerMode.DISABLED
                         ),
                     ): SelectSelector(
-                        SelectSelectorConfig(options=BLE_SCANNER_OPTIONS),
+                        SelectSelectorConfig(
+                            options=BLE_SCANNER_OPTIONS,
+                            translation_key=CONF_BLE_SCANNER_MODE,
+                        ),
                     ),
                 }
             ),

--- a/homeassistant/components/shelly/strings.json
+++ b/homeassistant/components/shelly/strings.json
@@ -71,5 +71,14 @@
     "abort": {
       "ble_unsupported": "Bluetooth support requires firmware version {ble_min_version} or newer."
     }
+  },
+  "selector": {
+    "ble_scanner_mode": {
+      "options": {
+        "disabled": "Disabled",
+        "active": "Active",
+        "passive": "Passive"
+      }
+    }
   }
 }

--- a/homeassistant/components/shelly/translations/en.json
+++ b/homeassistant/components/shelly/translations/en.json
@@ -71,5 +71,14 @@
                 "description": "Bluetooth scanning can be active or passive. With active, the Shelly requests data from nearby devices; with passive, the Shelly receives unsolicited data from nearby devices."
             }
         }
+    },
+    "selector": {
+        "ble_scanner_mode": {
+            "options": {
+                "disabled": "Disabled",
+                "active": "Active",
+                "passive": "Passive"
+            }
+        }
     }
 }

